Updates to Bootstrap 4.5 (with new date-time picker), Jquery 3.5.1, Moment 2.27, ui-bootstrap 0.12.0 and Media Display Features

limmabean opened the pull request:
Major overhauls in SCSS, templates, and angular controllers with a few changes in directives. This also includes media content additions (thumbnail and content). This also includes several updates that were made by Max to add API endpoints for SMTC tags.

This passes 219 test cases and fails 11. There are more failed tests in the develop branch (14).

This does not address a problem with will be fixed in a different branch: the show page of reports (when you click the content of a report) does not allow you to flag or add a report to an incident. This also does not include the tagging functionality. This will be accomplished in a different branch. 

I attempted to use BEM modifiers with all new CSS classes and remove as many uses of old classes as I could. The old CSS classes still largely remain, so some cleanup will be done soon after as well. If we could use BEM naming conventions from now on, it would be helpful.

[Added by Kevin:] BEM http://getbem.com/introduction/

The UI-bootstrap library has been modified and must be kept. This is because 0.12.0 does not support Bootstrap 4.5 but newer UI-bootstrap libraries require newer versions of angularjs. If we switch to angular 1.4 or higher, we can avoid using custom library files. 

There are still aria labels that need to be filled out, but there are no visual problems I can find.

karpawich committed:
feat: add SMTCTag model for report-level tagging

karpawich committed:
test: SMTC model

karpawich committed:
test: add SMTC model to general db setup & teardown routines

karpawich committed:
feat: expose API endpoints for SMTC tags

karpawich committed:
test: API endpoints for SMTC tags

Commit:
Work on updating Bootstrap to 4.5.0

This requires edits on templates and scss files.

Commit:
Fixing containers and adding new Jquery

Commit:
Merge branch 'develop' into ML-tag-frontend

Commit:
Fixed Substring Matches

This is the fix to substring matching for author and keyword. Tags still require full string match but that isn't problematic.

Commit:
Move to BootStrap 4

Some more updates to commit.

Commit:
Updates to DateTimePicker and Modal

mkyaw6 committed:
Add stats to report template

Commit:
Updated DateTimePicker and added stats

Commit:
Merge remote-tracking branch 'origin/333/report-level-tags' into ML-tag-frontend

Commit:
Moved Statsbar

Commit:
Merge branch 'develop' into ML-tag-frontend

Commit:
Updated to include Media on Show Page

Commit:
Switched to CDN and Edited Report Show View

Commit:
Upgrade to Loading Bar .8

Commit:
Moving Everything to the Card system. Added Stats to most Pages

Commit:
Woops I made the scss compile break

Commit:
Adding stats to Analysis

Commit:
Got rid of old bootstrap css and added stats to Settings Page

Commit:
Fixed Source Toggle

mkyaw6 committed:
Fix source icons not displaying

mkyaw6 committed:
Merge branch 'ML-tag-frontend' of https://github.com/TID-Lab/aggie into ML-tag-frontend
Merge remote changes to local

Commit:
Adjusted Tables to only have borders on their bodies

Also made the incidents a link when using the report incident modal.

Commit:
Merge branch 'ML-tag-frontend' of https://github.com/TID-Lab/aggie into ML-tag-frontend

mkyaw6 committed:
Change filter bar layout

mkyaw6 committed:
Fix Settings dropdown being behind stats

mkyaw6 committed:
Fix Incidents filter bar and center

mkyaw6 committed:
Fix navbar highlight

mkyaw6 committed:
Finish fixing navbar

Commit:
Fixed Layout Problems

Commit:
Final Bootstrap Push

Commit:
Small Responsive Table Fix

Commit:
Change some test configs to update to Jquery 3.5.1

Commit:
If there is not go button, url doesn't change

Commit:
Remove Tagging for now.

Commit:
Revert "test: SMTC model"

This reverts commit d10886d7eed9fd42e10898a7876d3bfd3eed77c5.

Commit:
Revert "test: API endpoints for SMTC tags"

This reverts commit 84007671b80004040fe1b342cd07e84c90e2fc3e.

Commit:
Revert "feat: expose API endpoints for SMTC tags"

This reverts commit c8b0f4cc5e3b00e9f10af891aa2303428d5aff02.

Commit:
Revert "feat: add SMTCTag model for report-level tagging"

This reverts commit 6ae4601c

Commit:
Revert "test: add SMTC model to general db setup & teardown routines"

This reverts commit 41492d49

limmabean commented:
I removed the SMTC commits for clairity.

Commit:
Updates to Statsbar responsiveness

I also added back functionality to the show templates.

Commit:
Fixed Show View for Incidents

Commit:
Added some missing alerts

Also fixed a modal input length.

cooperka committed:
Commit auto-changes to package-lock after installing

cooperka reviewed (commented):
MASSIVE improvements 👏 

I tested for usability/bugs, and only glanced at the code changes which should have more thorough review by someone else. A couple thoughts (fixed!):
- ~~The statsbar floating on the right goes away entirely with smaller screen sizes; could it instead go at the top or bottom so it's not lost?~~
- ~~Incident view pages e.g. http://aggie.test/incidents/5f171dad1771033da8b7b174 seem to have wonky styles, they're the only page not consistent with everything else (the statsbar is rendered at the bottom and the spacing is weird)~~

cooperka commented on public/angular/templates/reports/show.html:
```diff
@@ -1,47 +1,242 @@
+              <tr>
+                <th scope="row" class="detail-key" translate>Author</th>
+                <td colspan="10" class="detail-value">
+                  <a ng-href="{{report.metadata.accountUrl}}" target="_blank">{{ report.author }}</a>
```
This link looks clickable but **does nothing** on click for me:

<img width="365" alt="Screen Shot 2020-07-30 at 09 44 23" src="https://user-images.githubusercontent.com/2047062/88950371-68bec600-d249-11ea-825b-4b2c1d17da9a.png">

limmabean replied:
Fixed in https://github.com/TID-Lab/aggie/commit/899a45fee7d0c0358cd6288564c7fa60b37ecfe3. Also added views for twitter posts.

limmabean replied:
> * Incident view pages e.g. http://aggie.test/incidents/5f171dad1771033da8b7b174 seem to have wonky styles, they're the only page not consistent with everything else (the statsbar is rendered at the bottom and the spacing is weird)

Fixed with https://github.com/TID-Lab/aggie/commit/40d320741d82f098f25e9864f9bea898f2ad7f13.

cooperka commented on public/angular/templates/reports/show.html:
```diff
@@ -1,47 +1,242 @@
+                <button type="button" class="btn btn-danger" ng-click="toggleFlagged(true)">
+                  <span class="mr-2" translate> Flag </span>
+                  <i class="fa fa-flag" aria-hidden="true"></i>
+                </button>
```
Flag/unflag seems to have no effect here on the report show page

limmabean replied:
At the time of the PR they were placed there as placeholders because the functionality didn't actually exist on the original aggie. It is now done. See https://github.com/TID-Lab/aggie/commit/0b02c9e7fda490b0e673049feedd484364241a03 for the fix.

Commit:
fix: Clickable Author Link fixed and Content Service Show Views

I created a new folder for content-services so the show screen can display as much custom information as possible for each content service. Currently only implemented Crowdtangle and twitter.

Commit:
Feat: Added Favicon

I just recently realized that aggie has no favicon. I have added one.

Commit:
Added Linked Incident and Flagging Capability to Show Screen

This capability didn't exist before for some reason. I added it.

Commit:
Updated favicon to include transparency.

Commit:
Added Statsbar to User Pages

Commit:
Center sources.

limmabean commented:
Toggling should be a single button like bold or italicize in word.

Commit:
feat: Statsbar Moves to the Bottom upon reaching page Breakpoint

limmabean commented:
> * The statsbar floating on the right goes away entirely with smaller screen sizes; could it instead go at the top or bottom so it's not lost?

Addressed in https://github.com/TID-Lab/aggie/commit/ab6d7a404e0e91f175d817990b702d0fcdf3cf5c.

mkyaw6 commented on config/passphrase.txt:
```diff
@@ -0,0 +1 @@
+32guns
```
This seems to be the passphrase previously needed for PEM and should not be included in the PR

mkyaw6 committed:
Delete passphrase.txt

mkyaw6 committed:
Merge branch 'ML-tag-frontend' of https://github.com/TID-Lab/aggie into ML-tag-frontend

Commit:
Fixed Misleading Label for Reports/Min -> Reports/Last Min

Commit:
Changed to config Page

cooperka committed:
Update docs to reflect 'configuration' name

cooperka commented:
I pushed a tweak mainly updating docs; looks great!

Commit:
Merge branch 'develop' into ML-tag-frontend
